human-viz-bridge/README.md

```diff
@@ -1,9 +1,69 @@
 ## Tools for interfacing HDE module architecture with ROS RViz.
 
-In order to launch urdf models, possible related publishers and rviz itself, launch files included in misc folder have to be employed.
-To launch these files use this command (each time for each launch file):
+In order to visualize but also to assess the results of the HDE module architecture, some YARP modules that take as input Xsens system, human state, and human dynamics data and publish them in the form of ros topics (so as to be read from Rviz) have been implemented.
 
+Information about RViz can be found here: [wiki.ros](http://wiki.ros.org/rviz).
+
+Information about how using yarp with ROS can be found here: [yarp with ROS](http://www.yarp.it/yarp_with_ros.html)   
+
+The I/O scheme of these modules is the following:
+
+<img src="/misc/human-viz-bridge.png">
+
+Each module is described in detail below.
+
+- **HumanTFBridge** is the module in charge of reading from the Xsens system YARP interface data regarding the position and the orientation in the global reference frame of each human segment and publishing them in the form of ros topic. 
+The workflow is the following:
+	- the urdf model of the human is loaded and a publisher to the topic "/tf" is opened;
+	- since the urdf model we use includes "real" and "fake" segments to deal with joints characterised by more degreees of freedom, a list of the "real" segments is read from the module configuration file and put in a vector; then, comparing the list of all the segments taken from the urdf with the former list also a vector of the "fake" segments is created;
+	- a class tf2_msgs_TFMessage compatible with ROS is created containing the coordinate frame transforms for each segment of the urdf model;
+	- for the fake segments the class is filled with an identity transform since these segments are not visible in RViz while for the real segments is filled with data coming from the Xsens system YARP interface. These transforms are updated by means of a callback on a class BufferedPort < XsensSegmentsFrame > of the Xsens data and then published by means of a Publisher < tf2_msgs_TFMessage > class.
+NB: the module period is set to 100 since the data updating is controlled by the callback.
```

Comment:
100 what? Please always include the unit of measurements when you refer to a numerical quantity.